core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -901,12 +901,26 @@ trait OptimisticTransactionImpl extends TransactionalWrite
     // Post stats
     var numAbsolutePaths = 0
     val distinctPartitions = new mutable.HashSet[Map[String, String]]
-    val adds = actions.collect {
+
+    var bytesNew: Long = 0L
```

Comment:
this is a lot of imperative code added here. can you add some inline docs explaining what this is doing and why it is like this?